### ISSUE TITLE
Fix the errors due to change to Pearsonr function in scipy.stats

### ISSUE THREAD

**Issue by aloukina** [EducationalTestingService/rsmtool#343]
The function for computing Pearsonr in scipy.stats changed (https://github.com/scipy/scipy/commit/1acf46f508afa2c6d498e1001ca17e8ad98b46ef) which leads to two issues in RSMTool:

(1) Perfect correlation in some cases is now a value slightly less than  1. 
The new formula is: 
r = dot(xm/norm(xm), ym/norm(ym))
 
where
xm = x - mean(x)
ym = y - mean(y)
norm(xm) = sqrt(sum(abs(xm)^2))
 
This was done to resolve a number of issues including the fact that sqrt((xm*xm).sum()) in the old computation could not handle very large values such as [-5e210, 5e210, 3e200, -3e200]. (I checked and my version of R also returns NaN in this case).
 
The new implementation produces the same values with one exception: since the square root in the denominator is now computed before multiplication, the perfect correlations are (-)0.9999999999999998 instead of (-)1.0.  

(2) A ValueError is raised if N==1. 

These two changes lead to tests failing or returning errors (#339 ).

**Comment by aloukina** [comment]
We ran further comparisons using the following code as an example:
``` 
r=[]                                                                   
 
for i in range(10000): 
    u=np.random.uniform(size=100) 
    v=5+7*u 
    corr = pearsonr(u, v) 
    r.append(corr[0]) 
                                                                           
 len([v for v in r if v==1])      
```

For scipy 1.4 ~6,900/10,000 correlations equalled 0
For scipy 1.1 ~6,800/10,000 correlations equalled 0

Other implementations:
When using  `np.corrcoef` ~7,500/10,000 correlations equalled 0.
In `R` (see code below):  ~8,300 correlations out of 10,000 equalled 1.

In short, none of the standard implementations guarantee r==1 when correlation is by definition perfect. 
It's unfortunate that the new implementation fails the trivial case `[1, 2, 3]`. 

```
r=c()
 
for (i in seq(10000))
{
  u = runif(100)
  v = 5 +7*u
  um = u-mean(u)
  vm = v - mean(v)
  pr = cor(u, v)
  r=append(r, pr)
}
 
sum(r==1)
```

**Comment by desilinguist** [comment]
Addressed by #349.
